scripts/generateMetadata.js

```diff
@@ -46,32 +53,55 @@ async function docgen(filePath) {
   }
 }
 
-const getComponentNameFromFile = (file) => file.replace(/^.*[\\/]/, '').replace(/\.js/, '');
+const getComponentNameFromFile = (file) => path.basename(file).replace(/\.js/, '');
 
-// Get files recursively as an array
-const getFilesFromDirectory = (directoryPath) => {
-  const filesInDirectory = fs.readdirSync(path.join(root, directoryPath));
-  const files = filesInDirectory.map((file) => {
-    const filePath = path.join(directoryPath, file);
-    const stats = fs.statSync(path.join(root, filePath));
+/** Analyzes subcomponents of a component in given path and extracts paths they're imported from. */
+function getSubcomponentPaths(componentPath) {
+  const componentName = getComponentNameFromFile(componentPath);
 
-    if (stats.isDirectory()) {
-      return getFilesFromDirectory(filePath);
-    }
+  /** Matches lines that are like `Component.Subcomponent = Subcomponent` */
+  const subcomponentRegExp = new RegExp(
+    `(${componentName}\\.[A-Z]\\w+) = (?<subcomponentName>\\w+)`,
+    'g',
+  );
 
-    return filePath;
-  });
+  const fileContent = fs.readFileSync(path.join(root, componentPath), 'utf-8');
+  const subcomponentNameMatches = fileContent.matchAll(subcomponentRegExp);
+  const subcomponentNames = [...subcomponentNameMatches].map((a) => a.groups.subcomponentName);
+
+  if (!subcomponentNames.length) return [];
+
+  /** Matches lines that are like `Subcomponent from './path/to/subcomponent';` */
+  const subcomponentPathRegExp = new RegExp(
+    `(${subcomponentNames.join('|')}) from '(?<path>[.\\w\\/-]+)';`,
+    'g',
+  );
+
+  const subcomponentPathMatches = fileContent.matchAll(subcomponentPathRegExp);
+  const subcomponentPaths = [...subcomponentPathMatches].map((match) =>
+    path.join(componentPath, '../', match.groups.path),
+  );
 
-  return files.filter((file) => file.length).flat();
-};
+  return subcomponentPaths;
+}
+
+/**
+ * Gets components that are imported in `index.js` of given directory (assuming they're all exported).
+ */
+function getExposedFilesFromDirectory(directoryPath) {
```

Comment:
What benefit is this providing us over the previous approach?  Can you add in the PR description

Reply:
Basically, we're generating metadata only for the components imported in `index.js` of the packages. Previously we generated it for all the components, even for the docs site components, css and test files, which are unecessary. This was not what I actually wanted to do. I wanted to get rid of the DocGen errors polluting the terminal. So, I came up with this approach.

Reply:
I updated the PR title. What do you think?

Reply:
No the actual summary, what's the high-level overview of how this algo works?